airflow/serialization/serializers/datetime.py

```diff
@@ -44,7 +44,7 @@ def serialize(o: object) -> tuple[U, str, int, bool]:
         if is_naive(o):
             o = convert_to_utc(o)
 
-        tz = o.tzname()
+        tz = o.tzinfo.name  # type: ignore
```

Comment:
Dumb solution which should work for different types of tzinfo

```suggestion
        from pendulum.tz.timezone import FixedTimezone, Timezone

        tzi = o.tzinfo
        if isinstance(tzi, FixedTimezone):
            tz = tzi.offset or "UTC"
        elif isinstance(tzi, Timezone):
            tz = tzi.name
        else:
            tz = int(o.utcoffset().total_seconds()) or "UTC"
```

Reply:
And just for a record, I don't think that we should use local imports for `datetime` and `pendulum` in this module because we import `airflow.utils.timezone` which import entire `datetime` and some modules from `pendulum` (at least Timezone/Datetime)